ecommerce/serializers.py

```diff
@@ -155,6 +159,49 @@ def get_coupons(self, instance):
             )
         ]
 
+    def get_data_consents(self, instance):
+        """ Get the DataConsentUser objects associated with the basket via coupon and product"""
+        data_consents = []
+        coupon_selections = models.CouponSelection.objects.filter(basket=instance)
+        if coupon_selections:
+            courses = [
+                course
+                for courselist in [
```

Comment:
Can we move the list here outside of the list comprehension? Since there is no use of `course` inside this comprehension we should be able to only evaluate it once.